lib/src/mapbox_map.dart

```diff
@@ -580,14 +580,16 @@ class MapboxMap extends ChangeNotifier {
 
   /// References for all interactions added to the map.
   @experimental
-  final _InteractionsMap _interactionsMap =
-      _InteractionsMap(interactions: {});
+  final _InteractionsMap _interactionsMap = _InteractionsMap(interactions: {});
 
-  /// Add an interaction
+  /// Add an interaction to the map
+  /// An identifier can be provided, which you can use to remove
+  /// the interaction with `.removeInteraction(interactionID)`
   @experimental
   void addInteraction<T extends TypedFeaturesetFeature<FeaturesetDescriptor>>(
-      TypedInteraction<T> interaction) {
-    var id = _interactionsMap.interactions.length;
+      TypedInteraction<T> interaction,
+      {String? interactionID}) {
+    var id = interactionID ?? DateTime.now().microsecondsSinceEpoch.toString();
```

Comment:
@evil159 - Is there a better way to assign a unique string?

Reply:
Can we use UniqueKey here?https://api.flutter.dev/flutter/foundation/UniqueKey-class.html

Reply:
👍  -- updated in latest commit